### PR TITLE
Update add_dynamic_search_ads example

### PR DESCRIPTION
**Pull request by BenRKarl**
We already had an existing example for `add_dynamic_search_ads` that was misnamed, here I rename it to be consistent with other libraries as well as update some comments for consistency and update some syntax.

### CODE REVIEW

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 1]
```diff
@@ -41,16 +41,15 @@ def main(client, customer_id):
         ad_group_resource_name = create_ad_group(client, customer_id,
                                                  campaign_resource_name)
         create_expanded_dsa(client, customer_id, ad_group_resource_name)
-        add_webpage_criteria(client, customer_id, ad_group_resource_name)
+        add_webpage_criterion(client, customer_id, ad_group_resource_name)
     except GoogleAdsException as ex:
-        print('Request with ID "{}" failed with status "{}" and includes the '
-              'following errors:'.format(ex.request_id, ex.error.code().name))
+        print(f'Request with ID "{ex.request_id}" failed with status '
+              f'"{ex.error.code().name}" and includes the following errors:'
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: syntax error (missing final ')')

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 2]
```diff
@@ -41,16 +41,15 @@ def main(client, customer_id):
...
         for error in ex.failure.errors:
-            print('\tError with message "{}".'.format(error.message))
+            print(f'\tError with message "{error_message}".'))
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: syntax error (extra ')')

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 3]
```diff
@@ -64,12 +63,12 @@ def create_budget(client, customer_id):
...
     """
-    # Retrieve a new campaign budget operation object.
+    # Creates a campaign budget operation.
     campaign_budget_operation = client.get_type('CampaignBudgetOperation',
                                                 version='v3')
-    # Create a campaign budget.
+    # Issues a mutate request to add campaign budgets.
     campaign_budget = campaign_budget_operation.create
-    campaign_budget.name.value = 'Interplanetary Cruise #{}'.format(uuid4())
+    campaign_budget.name.value = f'Interplanetary Cruise #{uuid4()}')
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: syntax error (extra ')')

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 4]
```diff
@@ -154,31 +151,31 @@ def create_ad_group(client, customer_id, campaign_resource_name):
...
-    # Submit the ad group ad operation to add an ad group.
+
+    # Issues a mutate request to add the ad group.
     response = ad_group_service.mutate_ad_groups(customer_id,
                                                  [ad_group_operation])
     resource_name = response.results[0].resource_name
 
-    # Display the results.
-    print('Created Ad Group with resource_name: {}'.format(resource_name))
+    print(f'Created Ad Group with resource_name: {resource_name}'))
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: syntax error (extra ')')

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 5]
```diff
@@ -207,12 +204,11 @@ def create_expanded_dsa(client, customer_id, ad_group_resource_name):
                                                     [ad_group_ad_operation])
     resource_name = response.results[0].resource_name
 
-    # Display the results.
-    print('Created Ad Group Ad with resource_name: {}'.format(resource_name))
+    print(f'Created Ad Group Ad with resource_name: {resource_name}'))
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: syntax error (extra ')')

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 6]
```diff
@@ -245,14 +241,12 @@ def add_webpage_criteria(client, customer_id, ad_group_resource_name):
...
-    # Submit the ad group criterion operation to add the ad group criterion.
+    # Issues a mutate request to add the ad group criterion.
     response = ad_group_criterion_service.mutate_ad_group_criteria(
         customer_id, [ad_group_criterion_operation])
     resource_name = response.results[0].resource_name
 
-    # Display the results.
-    print('Created Ad Group Criterion with resource_name: {}'.format(
-        resource_name))
+    print(f'Created Ad Group Criterion with resource_name: {resource_name}'))
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: syntax error (extra ')')

**Comment by wihl** [comment]
Pls update copyright to 2020

**Comment by wihl** [comment]
In general, add "" around resource names when printing

**Comment by fiboknacky** [comment]
> Pls update copyright to 2020

Technically, it's the same example. Update of copyright is not required.

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 7]
```diff
@@ -207,12 +204,11 @@ def create_expanded_dsa(client, customer_id, ad_group_resource_name):
...
 
-    # Display the results.
-    print('Created Ad Group Ad with resource_name: {}'.format(resource_name))
+    print(f'Created Ad Group Ad with resource_name: {resource_name}'))
 
 
-def add_webpage_criteria(client, customer_id, ad_group_resource_name):
-    """Creates a Web Page Criteria to the given Ad Group.
+def add_webpage_criterion(client, customer_id, ad_group_resource_name):
+    """Creates a web page criteria to the given ad group.
```
Comment (fiboknacky) [inline examples/advanced_operations/add_dynamic_search_ads.py]: "web page criterion"

**Comment by BenRKarl** [comment]
@wihl @fiboknacky thanks for the feedback, I made updates and ran the example to make sure it works correctly.

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 8]
```diff
@@ -41,16 +41,15 @@ def main(client, customer_id):
...
-        add_webpage_criteria(client, customer_id, ad_group_resource_name)
+        add_webpage_criterion(client, customer_id, ad_group_resource_name)
     except GoogleAdsException as ex:
-        print('Request with ID "{}" failed with status "{}" and includes the '
-              'following errors:'.format(ex.request_id, ex.error.code().name))
+        print(f'Request with ID "{ex.request_id}" failed with status '
+              f'"{ex.error.code().name}" and includes the following errors:')
         for error in ex.failure.errors:
-            print('\tError with message "{}".'.format(error.message))
+            print(f'\tError with message "{error_message}".')
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: s/error_message/error.message/

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 9]
```diff
@@ -187,7 +184,7 @@ def create_expanded_dsa(client, customer_id, ad_group_resource_name):
     """
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: s/campaign_resource_name/ad_group_resource_name

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 10]
```diff
@@ -207,12 +204,11 @@ def create_expanded_dsa(client, customer_id, ad_group_resource_name):
                                                     [ad_group_ad_operation])
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: should probably unindent a bunch of spaces. The current indentation is inconsistent

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 11]
```diff
@@ -207,12 +204,11 @@ def create_expanded_dsa(client, customer_id, ad_group_resource_name):
...
+    print(f'Created Ad Group Ad with resource_name: "{resource_name}"')
 
 
-def add_webpage_criteria(client, customer_id, ad_group_resource_name):
-    """Creates a Web Page Criteria to the given Ad Group.
+def add_webpage_criterion(client, customer_id, ad_group_resource_name):
+    """Creates a web page criterion to the given ad group.
 
     Args:
         client: an initialized GoogleAdsClient instance.
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: s/campaign_resource_name/ad_group_resource_name

File: examples/advanced_operations/add_dynamic_search_ads.py [hunk 12]
```diff
@@ -230,13 +226,13 @@ def add_webpage_criteria(client, customer_id, ad_group_resource_name):
     # Optional: set the status.
     criterion.status = client.get_type(
         'AdGroupCriterionStatusEnum', version='v3').PAUSED
-    criterion.webpage.criterion_name.value = 'Special Offers'
 
+    # Sets the criterion to match a specific page URL and title.
```
Comment (wihl) [inline examples/advanced_operations/add_dynamic_search_ads.py]: nit: trailing whitespace

**Pull request merged**